### PR TITLE
Struct archi

### PR DESCRIPTION
**Pull request by alexandrelagasse**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,3 +32,4 @@
 *.app
 .clang-format
 .idea/
+cmake-build-debug/
```
Comment (romainpanno) [inline .gitignore:35]: EOF

File: src/Raytracer/CMakeLists.txt [hunk 2]
```diff
@@ -5,3 +5,5 @@ target_include_directories(raytracer PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}")
 target_sources(raytracer PRIVATE
     "${CMAKE_CURRENT_SOURCE_DIR}/Main.cpp"
 )
+
+add_subdirectory(Core)
```
Comment (romainpanno) [inline src/Raytracer/CMakeLists.txt:9]: EOF

File: src/Raytracer/Entity/ALight.hpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    class ALight : public ILight, public AEntity {
+    public:
+        ALight();
+
+        ~ALight();
+
+    protected:
+    private:
+    };
+};
```
Comment (romainpanno) [inline src/Raytracer/Entity/ALight.hpp:23]: EOF

File: src/Raytracer/Entity/EntityManager.hpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+        ~EntityManager();
+
+        void addEntity(IEntity &entity);
+
+        int getEntityCount() const;
+
+        IEntity &getEntity(int index) const;
+
+        IEntity &getEntity(std::string name) const;
```
Comment (romainpanno) [inline src/Raytracer/Entity/EntityManager.hpp:28]: const std::string &

File: src/Raytracer/Entity/ICam.hpp [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    class ICam {
+    public:
+        ICam();
+
+        ~ICam();
+
+    protected:
+    private:
+    };
+};
```
Comment (romainpanno) [inline src/Raytracer/Entity/ICam.hpp:22]: EOF

File: src/Raytracer/Entity/Scenes.hpp [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    class Scenes {
+    public:
+        Scenes();
+
+        ~Scenes();
+
+    protected:
+    private:
+    };
+};
```
Comment (romainpanno) [inline src/Raytracer/Entity/Scenes.hpp:20]: EOF

File: src/Raytracer/Entity/ScenesManager.hpp [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+        Scenes &getScene(int index) const;
+
+        Scenes &getScene(std::string name) const;
+
+    protected:
+    private:
+        std::vector<std::reference_wrapper<Scenes>> _scenes;
+    };
+};
```
Comment (romainpanno) [inline src/Raytracer/Entity/ScenesManager.hpp:33]: EOF

File: src/Raytracer/Plugin/LoadPlugin.hpp [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class LoadPlugin {
+public:
+    LoadPlugin();
+
+    ~LoadPlugin();
+
+protected:
+private:
```
Comment (romainpanno) [inline src/Raytracer/Plugin/LoadPlugin.hpp:18]: you need to indent

**Review by romainpanno — CHANGES_REQUESTED** [review]

**Review by KitetsuK — APPROVED** [review]

File: src/Raytracer/Entity/ScenesManager.hpp [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        ScenesManager();
+        ~ScenesManager();
+
+        void addScene(Scenes &scene);
+
+        int getSceneCount() const;
+
+        Scenes &getScene(int index) const;
+
+        Scenes &getScene(std::string name) const;
```
Comment (romainpanno) [inline src/Raytracer/Entity/ScenesManager.hpp:27]: const string &

**Review by romainpanno — CHANGES_REQUESTED** [review]

**Pull request merged**